### PR TITLE
aws - s3 - add support for storage lens configuration

### PR DESCRIPTION
**Pull request by jerryhxu**
closes (https://github.com/cloud-custodian/cloud-custodian/issues/9246)

This PR adds a new resource type s3-storage-lens, related tagging functionality and delete action.

### CODE REVIEW

File: c7n/resources/s3control.py [hunk 1]
```diff
@@ -95,3 +97,160 @@ class resource_type(TypeInfo):
...
+                  key: test
+                  value: storagelens
+    """
+    permissions = ('s3:GetStorageLensConfigurationTagging',
+                   's3:PutStorageLensConfigurationTagging',)
+
+    def process_resource_set(self, client, resources, add_tags):
+        accountId=self.manager.config.account_id
+        for r in resources:
+            configId=r['StorageLensConfiguration']['Id']
```
Comment (kapilt) [inline c7n/resources/s3control.py:160]: nit, its unfortunate that ruff doesn't pick this up, but please put spaces between operators re `x = 1`

Comment (jerryhxu) [reply c7n/resources/s3control.py]: sure, it's updated.

**Comment by kapilt** [comment]
it was a little unclear if resource group tagging api supports s3 lens, its listed there, if so it would obviate the need for some of the tag implementation here.

**Comment by jerryhxu** [comment]
@kapilt , resource group tagging api does not support storagelens. I tested it and it didn't work. Also see this aws doc: [
](https://docs.aws.amazon.com/ARG/latest/userguide/supported-resources.html), if you look for storagelens, it is not supported.

**Comment by kapilt** [comment]
> @kapilt , resource group tagging api does not support storagelens. I tested it and it didn't work. Also see this aws doc: (https://docs.aws.amazon.com/ARG/latest/userguide/supported-resources.html), search for storagelens, it is not supported.

I tried it just now and it does seem to work

```
❯ aws resourcegroupstaggingapi get-resources --resource-arn-list arn:aws:s3:us-east-1:xxxx:storage-lens/default-account-dashboard
{
    "ResourceTagMappingList": [
        {
            "ResourceARN": "arn:aws:s3:us-east-1:xxxx:storage-lens/default-account-dashboard",
            "Tags": [
                {
                    "Key": "Env",
                    "Value": "Dev"
                }
            ]
        }
    ]
}
```

**Comment by jerryhxu** [comment]
@kapilt, thanks for the comment! I made an error before. I just updated the PR to reflect the change. That greatly simplified the tagging code.

File: c7n/resources/s3control.py [hunk 2]
```diff
@@ -95,3 +97,89 @@ class resource_type(TypeInfo):
...
+
+    def augment(self, resources):
+        client = local_session(self.manager.session_factory).client('s3control')
+        results = []
+        for r in resources:
+            arn = Arn.parse(r['StorageLensArn'])
+            storage_lens_configuration = client \
+                .get_storage_lens_configuration(AccountId=arn.account_id, ConfigId=r['Id'])\
+                .get('StorageLensConfiguration')
+            tags = client \
```
Comment (kapilt) [inline c7n/resources/s3control.py:116]: use `universal_augment` which can fetch bulk fetch tags for multiple resources instead of fetching per resource

**Review by kapilt — APPROVED** [review]
one minor re augment, else lgtm

File: c7n/resources/s3control.py (hunk 2 above)
Comment (jerryhxu) [reply c7n/resources/s3control.py]: yeah, code updated using universal_augment.

File: c7n/resources/s3control.py [hunk 3]
```diff
@@ -95,3 +97,88 @@ class resource_type(TypeInfo):
...
+        client = local_session(self.manager.session_factory).client('s3control')
+        results = []
+        for r in resources:
+            storage_lens_configuration = self.manager.retry( \
+                client.get_storage_lens_configuration,
+                AccountId=self.manager.config.account_id,
+                ConfigId=r['Id']) \
+                .get('StorageLensConfiguration')
+            results.append(storage_lens_configuration)
+        return universal_augment(
```
Comment (PratMis) [inline c7n/resources/s3control.py:118]: We could just change it to 
```return universal_augment(self.manager, super().augment(results))```

Comment (jerryhxu) [reply c7n/resources/s3control.py:118]: @PratMis , sounds good. I simplified the code.

**Pull request merged**